src/include/sof/lib/dma.h

```diff
@@ -326,9 +326,10 @@ static inline int dma_stop(struct dma_chan_data *channel)
 static inline int dma_copy(struct dma_chan_data *channel, int bytes,
 			   uint32_t flags)
 {
-	int ret = channel->dma->ops->copy(channel, bytes, flags);
+	if (channel && channel->dma && channel->dma->ops && channel->dma->ops->copy)
+		return channel->dma->ops->copy(channel, bytes, flags);
```

Comment:
Overkill. If channel is not null, channel->dma and channel->dma->ops should not be null (you can make an assert() out of that), and I believe the copy operation may not be null either. So make an if (channel) { assert(channel->dma); assert(channel->dma->ops); assert(channel->dma->ops->copy); return ...} or so.

Reply:
Also possibly feels like it's not the right solution in the first place anyway.

Reply:
Good point, I haven't root caused yet which explicit item is NULL as the stress test issue is quite time consuming to reproduce, let me try to simplify it and put the check to the caller.